datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala

```diff
@@ -112,14 +111,13 @@ object DatasetComparisonJob {
     )
   }
 
-  private def writeMetricsToFile(result: ComparisonResult, fileName: String): Unit = {
-    new File(fileName).mkdirs()
-    val writer = new PrintWriter(s"$fileName/_METRICS")
-    try {
-      writer.write(result.getJsonMetadata)
-    } finally {
-      writer.close()
-    }
+  private def writeMetricsToFile(result: ComparisonResult, fileName: String)(implicit sparkSession: SparkSession): Unit = {
+    val path = new Path(s"$fileName/_METRICS")
+    val fs = FileSystem.get(sparkSession.sparkContext.hadoopConfiguration)
+
+    val v = fs.create(path)
```

Comment:
- `fs.create` can throw exceptions too.
- `v` is an odd name for a file output stream. You can use `out` or `fout`, for instance.

Reply:
> fs.create can throw exceptions too.

I might remember this wrongly but does it leave the stream open? If it fails in creation/opening it's "fine" and throws a proper error of not being able to.

Reply:
I agree that if the desired behavior is to pass through the exception if `.create()` throws than this implementation is completely correct.

Here is the Java approach to managing resources: https://www.baeldung.com/java-try-with-resources
But if I remember correctly this does not work in Scala. That's why @wajda created [ARMImplicits](https://github.com/AbsaOSS/atum/blob/master/atum/src/main/scala/za/co/absa/atum/utils/ARMImplicits.scala). 

But I didn't find its usage in Enceladus so probably we might not go that far,

Reply:
I just realised I have `usingFile` [here](https://github.com/AbsaOSS/hermes/blob/develop/utils/src/main/scala/za/co/absa/hermes/utils/FileReader.scala#L37), which seems the same, but I only use it for reading. Maybe let's chat in the office.